### PR TITLE
Additional Check To Verify Staging for email SFTP is disabled on Dev

### PR DESCRIPTION
**Pull request by abhijitvk**
## Description of change
Additional check to make sure that the environment is staging before sending staging recipients email

Related to issue: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16763

#### Unique to this PR
- [x] spool_submissions_report_mailer.rb and year_to_date_report_mailer.rb files updated to send staging recipient emails with an additional check

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

**Comment by desireeturner** [comment]
Covers changes in https://github.com/department-of-veterans-affairs/vets.gov-team/issues/16763

**Comment by abhijitvk** [comment]
@department-of-veterans-affairs/backend-review-group : Could I get a code review on this at your earliest convenience?

**Comment by abhijitvk** [comment]
> won't this send emails to va stakeholders from all the dev environments?

Good catch, thanks. I have additionally retained your email for the dev email notification. let me know if this looks fine to you and also acceptable for the notification part

**Comment by lihanli** [comment]
i don't need to receive emails from all the dev environments, how about just sending the emails in staging?

File: app/workers/education_form/create_daily_fiscal_year_to_date_report.rb [hunk 1]
```diff
@@ -234,7 +234,7 @@ def generate_csv
     def perform
       Sentry::TagRainbows.tag
       filename = generate_csv
-      return unless FeatureFlipper.send_email?
+      return unless FeatureFlipper.send_email && !Rails.env.development?
```
Comment (lihanli) [inline app/workers/education_form/create_daily_fiscal_year_to_date_report.rb]: the rails env for staging/dev is `production`, still need to use the hostname method

Comment (abhijitvk) [reply app/workers/education_form/create_daily_fiscal_year_to_date_report.rb]: ok, made it simpler now to use hostname check to not include `dev-api`

**Pull request merged**